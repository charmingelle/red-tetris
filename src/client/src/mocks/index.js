export const RED = {
  main: '#c00',
  lighter: '#f00',
  darker: '#980000',
};

export const LINE = [
  [0, 0, 1, 0],
  [0, 0, 1, 0],
  [0, 0, 1, 0],
  [0, 0, 1, 0],
];

export const GI = [
  [0, 0, 0],
  [1, 1, 1],
  [0, 0, 1],
];

export const EMPTY_PILE = [
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
];

export const NOT_EMPTY_PILE = [
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [RED, RED, RED, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, RED, 0, 0, 0, 0, 0, 0, 0],
];

export const GAME_OVER_PILE = [
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, RED, RED, RED, RED, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [RED, RED, RED, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, RED, 0, 0, 0, 0, 0, 0, 0],
];

export const PILE_WITH_ROWS_TO_REMOVE = [
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [RED, RED, RED, RED, RED, RED, RED, RED, RED, RED],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [RED, RED, RED, RED, RED, RED, RED, RED, RED, RED],
  [RED, RED, RED, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, RED, 0, 0, 0, 0, 0, 0, 0],
];

export const TETRO = {
  figure: [GI],
  row: 5,
  col: 4,
  color: RED,
};

export const ROOM = {
  id: 'Room',
  leader: 'id-0',
  players: {
    'id-0': {
      id: 'id-0',
      name: 'Anna',
      pile: EMPTY_PILE,
      tetro: null,
      score: 0,
      isGameOver: false,
    },
  },
};

export const EMPTY_STORE = {
  socket: null,
  myId: null,
  myName: null,
  myRoom: null,
  people: {},
  rooms: {},
  tetro: null,
};

export const PERSON_ANNA = { name: 'Anna', score: 0 };

export const PERSON_OLYA = { name: 'Olya', score: 0 };
