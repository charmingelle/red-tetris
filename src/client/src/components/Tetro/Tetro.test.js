import React from 'react';
import { Tetro } from './Tetro';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { RED } from '../../mocks';

const mockStore = configureStore([]);

test('Tetro fully visible', () => {
  const store = mockStore({
    socket: null,
    myId: null,
    myName: null,
    myRoom: null,
    people: {},
    rooms: {},
    tetro: {
      figure: [
        [0, 0, 0],
        [1, 1, 1],
        [0, 0, 1],
      ],
      row: 5,
      col: 4,
      color: RED,
    },
  });

  expect(
    renderer
      .create(
        <Provider store={store}>
          <Tetro />
        </Provider>,
      )
      .toJSON(),
  ).toMatchSnapshot();
});

test('Tetro with only bottom visible', () => {
  const store = mockStore({
    socket: null,
    myId: null,
    myName: null,
    myRoom: null,
    people: {},
    rooms: {},
    tetro: {
      figure: [
        [0, 0, 0],
        [1, 1, 0],
        [0, 1, 1],
      ],
      row: 1,
      col: 4,
      color: RED,
    },
  });

  expect(
    renderer
      .create(
        <Provider store={store}>
          <Tetro />
        </Provider>,
      )
      .toJSON(),
  ).toMatchSnapshot();
});
