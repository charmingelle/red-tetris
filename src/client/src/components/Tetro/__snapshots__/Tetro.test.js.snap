// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tetro fully visible 1`] = `
Array [
  <div
    className="square"
    style={
      Object {
        "backgroundColor": "#c00",
        "borderBottom": "5px solid #980000",
        "borderLeft": "5px solid #f00",
        "borderRight": "5px solid #980000",
        "borderTop": "5px solid  #f00",
        "height": "40px",
        "left": "160px",
        "top": "120px",
        "width": "40px",
      }
    }
  />,
  <div
    className="square"
    style={
      Object {
        "backgroundColor": "#c00",
        "borderBottom": "5px solid #980000",
        "borderLeft": "5px solid #f00",
        "borderRight": "5px solid #980000",
        "borderTop": "5px solid  #f00",
        "height": "40px",
        "left": "200px",
        "top": "120px",
        "width": "40px",
      }
    }
  />,
  <div
    className="square"
    style={
      Object {
        "backgroundColor": "#c00",
        "borderBottom": "5px solid #980000",
        "borderLeft": "5px solid #f00",
        "borderRight": "5px solid #980000",
        "borderTop": "5px solid  #f00",
        "height": "40px",
        "left": "240px",
        "top": "120px",
        "width": "40px",
      }
    }
  />,
  <div
    className="square"
    style={
      Object {
        "backgroundColor": "#c00",
        "borderBottom": "5px solid #980000",
        "borderLeft": "5px solid #f00",
        "borderRight": "5px solid #980000",
        "borderTop": "5px solid  #f00",
        "height": "40px",
        "left": "240px",
        "top": "160px",
        "width": "40px",
      }
    }
  />,
]
`;

exports[`Tetro with only bottom visible 1`] = `
Array [
  <div
    className="square"
    style={
      Object {
        "backgroundColor": "#c00",
        "borderBottom": "5px solid #980000",
        "borderLeft": "5px solid #f00",
        "borderRight": "5px solid #980000",
        "borderTop": "5px solid  #f00",
        "height": "40px",
        "left": "200px",
        "top": "0px",
        "width": "40px",
      }
    }
  />,
  <div
    className="square"
    style={
      Object {
        "backgroundColor": "#c00",
        "borderBottom": "5px solid #980000",
        "borderLeft": "5px solid #f00",
        "borderRight": "5px solid #980000",
        "borderTop": "5px solid  #f00",
        "height": "40px",
        "left": "240px",
        "top": "0px",
        "width": "40px",
      }
    }
  />,
]
`;
